### ISSUE TITLE
Ordinal Logical Type cause error to EvalML

### ISSUE THREAD

**Issue by iki77** [alteryx/evalml#2456]
I got an error when using Ordinal Logical Type with EvalML, the data I used is from [default credit card](https://archive.ics.uci.edu/ml/datasets/default+of+credit+card+clients)
I have defined the order values when I initiate it with woodowoek, but suddenly the order values are gone when used with EvalML.

```python
order = sorted(features["EDUCATION"].unique())
logical_types = {
    "AGE": "Age",
    "EDUCATION": Ordinal(order=order)
}
features.ww.init(logical_types=logical_types)

features_train, features_holdout, target_train, target_holdout = evalml.preprocessing.split_data(
    features, 
    target, 
    problem_type="binary",
    test_size=0.2,
    random_seed=42
)

model_families= ["linear_model", "random_forest"]

automl = evalml.AutoMLSearch(
    X_train=features_train, 
    y_train=target_train,
    problem_type="binary",
    objective="f1",
    max_batches=1,
    random_seed=42
)
automl.search()
```

The error log:
```
*****************************
* Beginning pipeline search *
*****************************

Optimizing for F1. 
Greater score is better.

Using SequentialEngine to train and score pipelines.
Searching up to 1 batches for a total of 4 pipelines. 
Will stop searching for new pipelines after 180 seconds.

Allowed model families: linear_model, random_forest

Evaluating Baseline Pipeline: Mode Baseline Binary Classification Pipeline
			Fold 0: Encountered an error.
			Fold 0: All scores will be replaced with nan.
			Fold 0: Please check d:\credit_default\notebooks\evalml_debug.log for the current hyperparameters and stack trace.
			Fold 0: Exception during automl search: Must use an Ordinal instance with order values defined
			Fold 1: Encountered an error.
			Fold 1: All scores will be replaced with nan.
			Fold 1: Please check d:\credit_default\notebooks\evalml_debug.log for the current hyperparameters and stack trace.
			Fold 1: Exception during automl search: Must use an Ordinal instance with order values defined
			Fold 2: Encountered an error.
			Fold 2: All scores will be replaced with nan.
			Fold 2: Please check d:\credit_default\notebooks\evalml_debug.log for the current hyperparameters and stack trace.
			Fold 2: Exception during automl search: Must use an Ordinal instance with order values defined
Mode Baseline Binary Classification Pipeline:
	Starting cross validation
	Finished cross validation - mean F1: nan

```

**Comment by chukarsten** [comment]
@iki77 we're working a solution to this and will target our next EvalML release, 0.28.0 for deployment.

**Comment by chukarsten** [comment]
@iki77 the fix for this should be released today with 0.29.0.  @dsherry , closing this.  @iki77, please reopen if you find this is still an issue.
